burn-dataset/src/dataset/sqlite.rs

```diff
@@ -218,4 +657,172 @@ mod tests {
+        assert!(storage.exists());
+        let result = storage.writer::<Sample>(true);
+        assert!(result.is_ok());
+    }
+
+    #[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+    pub struct Complex {
+        column_str: String,
+        column_bytes: Vec<u8>,
+        column_int: i64,
+        column_bool: bool,
+        column_float: f64,
+        colomn_complex: Vec<Vec<Vec<[u8; 3]>>>,
+    }
+
+    /// Create a temporary directory.
+    #[fixture]
+    fn tmp_dir() -> TempDir {
+        // Create a TempDir. This object will be automatically
+        // deleted when it goes out of scope.
+        tempdir().unwrap()
+    }
+    type Writer = SqliteDatasetWriter<Complex>;
+
+    /// Create a SqliteDatasetWriter with a temporary directory.
+    /// Make sure to return the temporary directory so that it is not deleted.
+    #[fixture]
+    fn writer_fixture(tmp_dir: TempDir) -> (Writer, TempDir) {
+        let temp_dir_str = tmp_dir.path();
+        let storage = SqliteDatasetStorage::from_name("preprocessed").with_base_dir(temp_dir_str);
+        let overwrite = true;
+        let result = storage.writer::<Complex>(overwrite);
+        assert!(result.is_ok());
+        let writer = result.unwrap();
+        (writer, tmp_dir)
+    }
+
+    #[test]
+    fn test_new() {
+        // Test that the constructor works with overwrite = true
+        let test_path = NamedTempFile::new().unwrap();
+        let result = SqliteDatasetWriter::<Complex>::new(&test_path, true);
+        assert!(result.is_ok());
+        assert!(!test_path.path().exists());
+
+        // Test that the constructor works with overwrite = false
+        let test_path = NamedTempFile::new().unwrap();
+        let result = SqliteDatasetWriter::<Complex>::new(&test_path, false);
+        assert!(result.is_err());
+
+        // Test that the constructor works with no existing file
+        let temp = NamedTempFile::new().unwrap();
+        let test_path = temp.path().to_path_buf();
+        assert!(temp.close().is_ok());
+        assert!(!test_path.exists());
+        let result = SqliteDatasetWriter::<Complex>::new(&test_path, true);
+        assert!(result.is_ok());
+        assert!(!test_path.exists());
+    }
+
+    #[rstest]
+    pub fn sqlite_writer_write(writer_fixture: (Writer, TempDir)) {
+        // Get the dataset_saver from the fixture and tmp_dir (will be deleted after scope)
+        let (writer, _tmp_dir) = writer_fixture;
+
+        assert!(writer.overwrite);
+        assert!(!writer.db_file.exists());
+
+        let new_item = Complex {
+            column_str: "HI1".to_string(),
+            column_bytes: vec![1_u8, 2, 3],
+            column_int: 0,
+            column_bool: true,
+            column_float: 1.0,
+            colomn_complex: vec![vec![vec![[1, 23_u8, 3]]]],
+        };
+
+        let result = writer.write("train", &new_item);
+
+        assert!(result.is_ok());
```

Comment:
I think you can remove those asserts in general, the `unwrap` will make the test crash. If we want more clarify, we could use expect with error messages.